src/index.ts

```diff
@@ -1,33 +1,31 @@
-/**
- * Welcome to Cloudflare Workers! This is your first worker.
- *
- * - Run `npm run dev` in your terminal to start a development server
- * - Open a browser tab at http://localhost:8787/ to see your worker in action
- * - Run `npm run deploy` to publish your worker
- *
- * Learn more at https://developers.cloudflare.com/workers/
- */
-
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface Env {
-  // Example binding to KV. Learn more at https://developers.cloudflare.com/workers/runtime-apis/kv/
-  // MY_KV_NAMESPACE: KVNamespace;
-  //
-  // Example binding to Durable Object. Learn more at https://developers.cloudflare.com/workers/runtime-apis/durable-objects/
-  // MY_DURABLE_OBJECT: DurableObjectNamespace;
-  //
-  // Example binding to R2. Learn more at https://developers.cloudflare.com/workers/runtime-apis/r2/
-  // MY_BUCKET: R2Bucket;
-  //
-  // Example binding to a Service. Learn more at https://developers.cloudflare.com/workers/runtime-apis/service-bindings/
-  // MY_SERVICE: Fetcher;
-  //
-  // Example binding to a Queue. Learn more at https://developers.cloudflare.com/queues/javascript-apis/
-  // MY_QUEUE: Queue;
-}
-
-export default {
-  async fetch (request: Request, env: Env, ctx: ExecutionContext): Promise<Response> {
-    return new Response('Hello World!')
-  }
-}
+import { Env } from "./env";
+import { Hono } from "hono";
+import { Service } from "./types";
+
+const app = new Hono<{ Bindings: Env }>();
+app
+  .all("*", async (c) => {
+    const xService = c.req.raw.headers.get("x-service");
+
+    if (!xService) {
+      return c.newResponse("x-service header is required", { status: 400 });
+    }
+
+    const service = await c.env.SERVICES.get<Service>(xService, { type: "json" });
+
+    if (!service) {
+      return c.newResponse("Service not found", { status: 500 });
+    }
+
+    console.log(service.host, service.token);
+
+    const url = new URL(service.host);
```

Comment:
Should this just be a header as well? `x-gateway-host`

Reply:
Certainly, it's possible, but I've already stored it in the Cloudflare (KV), so it's not essential. This approach enhances dynamism. If the host changes in the future, simply update the value in the Cloudflare KV via dashboard, and it will seamlessly function.

Reply:
Having it being in the header means I don't need to deploy anything if I want to add a host tho

Reply:
Alright, I'll utilize the `x-gateway-service-host` header for this.